src/visitable.cpp

```diff
@@ -797,10 +806,10 @@ int visitable<inventory>::amount_of( const std::string& what, bool pseudo, int l
 
     long res = 0;
     for( const item *it : iter->second ) {
-        res += it->amount_of( what, pseudo, limit );
+        res = accumulate_with_limit( res, it->amount_of( what, pseudo, limit ), limit );
```

Comment:
As above, this is now much more opaque what's going on.

Programmers become good at parsing `+`, `-` etc but not good at reading code such as this